extensions/amp-form/0.1/amp-form.js

```diff
@@ -260,17 +316,114 @@ function onInvalidInputBlur_(event) {
  * @param {!HTMLInputElement} input
  */
 function reportInputValidity(input) {
-  input./*OK*/focus();
+  if (isReportValiditySupported()) {
+    input.reportValidity();
+  } else {
+    input./*OK*/focus();
+
+    // Remove any previous listeners on the same input. This avoids adding many
+    // listeners on the same element when the user submit pressing Enter or any
+    // other method to submit the form without the element losing focus.
+    input.removeEventListener('blur', onInvalidInputBlur_);
```

Comment:
Instead of registering on the input, we could register a single listener on the form. Also, I hate this `#removeEventListener` stuff.